modulus/models/diffusion/preconditioning.py

```diff
@@ -780,6 +780,8 @@ def forward(
         c_in = 1 / (self.sigma_data**2 + sigma**2).sqrt()
         c_noise = sigma.log() / 4
 
+        x = torch.cat((x, img_lr), dim=1)
+
         F_x = self.model(
             (c_in * x).to(dtype),
```

Comment:
This looks incorrect. Is the latent `x` being scaled by `c_in` twice?

Suggest using this implementation, which is in my EDM code base

```
        if condition is None:
            arg = c_in * x
        else:
            condition = condition.to(torch.float32)
            arg = torch.cat([c_in * x, condition], dim=1)

        F_x = self.model(
            arg.to(dtype), c_noise.flatten(), class_labels=class_labels, **model_kwargs
        )
        assert F_x.dtype == dtype
        D_x = c_skip * x + c_out * F_x.to(torch.float32)
        return D_x
```

Reply:
easier to follow IMO without overwriting the name "x". That is why I instead name it `arg`.